backend-app/backend-app/UserRepository/UserServices.cs

```diff
@@ -87,45 +86,153 @@ public bool VerifyPassword(string hashedPassword, string providedPassword)
         return ComputeHash(providedPassword) == hashedPassword;
     }
 
+    public async Task<AuthTokens> GenerateJwtTokenLogin(User user)
+    {
+        var tokenHandler = new JwtSecurityTokenHandler();
+        var key = Encoding.UTF8.GetBytes(_configuration["Jwt:Key"]);
 
-    public string GenerateJwtTokenLogin(User user)
+        var claims = new List<Claim>
+        {
+            new Claim(ClaimTypes.NameIdentifier, user.Id.ToString()),
+            new Claim(ClaimTypes.Email, user.Email),
+        };
+
+        foreach (var userRole in user.UserRoles)
+        {
+            claims.Add(new Claim(ClaimTypes.Role, userRole.Role.Type));
+        }
+
+        claims.Add(new Claim(JwtRegisteredClaimNames.Iss, _configuration["Jwt:Issuer"]));
+        claims.Add(new Claim(JwtRegisteredClaimNames.Aud, _configuration["Jwt:Audience"]));
+
+        var accessTokenDescriptor = new SecurityTokenDescriptor
+        {
+            Subject = new ClaimsIdentity(claims),
+            Expires = DateTime.UtcNow.AddMinutes(2),
+            SigningCredentials = new SigningCredentials(new SymmetricSecurityKey(key), SecurityAlgorithms.HmacSha256Signature)
+        };
+
+        var accessToken = tokenHandler.CreateToken(accessTokenDescriptor);
+        var accessTokenString = tokenHandler.WriteToken(accessToken);
+
+        var existingRefreshToken = await GetExistingRefreshTokenAsync(user.Id);
+        string refreshToken;
+
+        if (existingRefreshToken != null && existingRefreshToken.ExpirationDate > DateTime.UtcNow)
+        {
+            refreshToken = existingRefreshToken.Token;
+        }
+        else
+        {
+            refreshToken = GenerateRefreshToken();
+            await StoreRefreshToken(refreshToken, user.Id, DateTime.UtcNow.AddMinutes(15));
```

Comment:
Are we generating refresh token only if refresh token does not exist?
But, as we discussed, a user can have multiple refresh token as they can login from multiple devices.

Reply:
Yes .. This " private async Task<RefreshToken> GetExistingRefreshTokenAsync(int userId)" if the token exists so no multiple refresh token is created . Yes we first check if the refresh token exist or not in the access token generation

Reply:
But, there can be multiple refresh tokens per user if the login/signup from different devices.